tests/utils/utils.go

```diff
@@ -0,0 +1,333 @@
+	teleporterContractAddress common.Address,
+) *types.Transaction {
+	nonce, err := source.ChainWSClient.NonceAt(ctx, fundedAddress, nil)
+	Expect(err).Should(BeNil())
+
+	data, err := teleporter.EVMTeleporterContractABI.Pack(
+		"sendCrossChainMessage",
+		SendCrossChainMessageInput{
+			DestinationChainID: destination.BlockchainID,
+			DestinationAddress: fundedAddress,
+			FeeInfo: FeeInfo{
+				ContractAddress: fundedAddress,
+				Amount:          big.NewInt(0),
+			},
+			RequiredGasLimit:        big.NewInt(1),
+			AllowedRelayerAddresses: []common.Address{},
+			Message:                 []byte{1, 2, 3, 4},
+		},
+	)
+	Expect(err).Should(BeNil())
+
+	// Send a transaction to the Teleporter contract
+	tx := types.NewTx(&types.DynamicFeeTx{
+		ChainID:   source.ChainIDInt,
+		Nonce:     nonce,
+		To:        &teleporterContractAddress,
+		Gas:       DefaultTeleporterTransactionGas,
+		GasFeeCap: DefaultTeleporterTransactionGasFeeCap,
+		GasTipCap: DefaultTeleporterTransactionGasTipCap,
+		Value:     DefaultTeleporterTransactionValue,
+		Data:      data,
+	})
+
+	txSigner := types.LatestSignerForChainID(source.ChainIDInt)
+	signedTx, err := types.SignTx(tx, txSigner, fundedKey)
+	Expect(err).Should(BeNil())
+
+	return signedTx
+}
+
+// Issues txs to activate the proposer VM fork on the specified subnet index in the manager
+func SetUpProposerVM(ctx context.Context, fundedKey *ecdsa.PrivateKey, manager *runner.NetworkManager, index int) {
+	subnet := manager.GetSubnets()[index]
+	subnetDetails, ok := manager.GetSubnet(subnet)
+	Expect(ok).Should(BeTrue())
+
+	chainID := subnetDetails.BlockchainID
+	uri := HttpToWebsocketURI(subnetDetails.ValidatorURIs[0], chainID.String())
+
+	client, err := ethclient.Dial(uri)
+	Expect(err).Should(BeNil())
+	chainIDInt, err := client.ChainID(ctx)
+	Expect(err).Should(BeNil())
+
+	err = utils.IssueTxsToActivateProposerVMFork(ctx, chainIDInt, fundedKey, client)
+	Expect(err).Should(BeNil())
+}
+
+// Blocks until all validators specified in nodeURIs have reached the specified block height
+func WaitForAllValidatorsToAcceptBlock(ctx context.Context, nodeURIs []string, blockchainID ids.ID, height uint64) {
+	for i, uri := range nodeURIs {
+		chainAWSURI := HttpToWebsocketURI(uri, blockchainID.String())
+		log.Info("Creating ethclient for blockchainA", "wsURI", chainAWSURI)
+		client, err := ethclient.Dial(chainAWSURI)
+		Expect(err).Should(BeNil())
+		defer client.Close()
+
+		// Loop until each node has advanced to >= the height of the block that emitted the warp log
+		for {
+			block, err := client.BlockByNumber(ctx, nil)
+			Expect(err).Should(BeNil())
+			if block.NumberU64() >= height {
+				log.Info("client accepted the block containing SendWarpMessage", "client", i, "height", block.NumberU64())
+				break
+			}
+		}
+	}
+}
+
+// Constructs and sends a transaction containing a warp message for the destination chain.
+// Returns the signed transaction.
+func ConstructAndSendWarpTransaction(
```

Comment:
I think we can likely make the functions in utils here a bit more modular and symmetric with each other.

Right now we have:
- `ConstructAndSendWarpTransaction` used for creating and sending a transaction that calls "receiveCrossChainMessage" of Teleporter. Returns the signed transaction and the tx receipt (though it looks like only the receipt is ever used currently)
- `CreateTeleporterTransaction` used for creating a transaction that calls "sendCrossChainMessage" of Teleporter.
- `SendTransactionAndWaitForAcceptance` used for sending any EVM transaction and waiting until it is included in a block.

There is duplicate logic for getting/setting the nonce to be used in transactions, and there is some gas price logic that differs between the utilities also.

I'd recommend breaking those up into:
- `CreateSendCrossChainMessageTransaction`
- `CreateReceiveCrossChainMessageTransaction`
- `SendTransactionAndWaitForAcceptance`

And having some un-exported helpers used by them:
- `setGasPriceAndNonce`
- `signTransaction`

Open to any other suggestions also; there is likely more improvement options available.

Reply:
Made these changes. These utilities have developed organically over the past few PRs, so appreciate you taking a step back and reevaluating their usefulness.